python/pyarrow/error.pyx

```diff
@@ -19,13 +19,52 @@ from pyarrow.includes.libarrow cimport CStatus
 from pyarrow.includes.common cimport c_string
 from pyarrow.compat import frombytes
 
+
 class ArrowException(Exception):
     pass
 
+
+class ArrowInvalid(ValueError, ArrowException):
```

Comment:
My preference is to only raise these exceptions for errors that originate from the C++ libraries, otherwise use the Python built-in ones. Sound good?